lib/attrsets.nix

```diff
@@ -444,6 +444,105 @@ rec {
     else
       [];
 
+  /* Recursively collect values produced by function `f` that verify a given
+     predicate named `pred` from the set `attrs`. The recursion is stopped
+     when the predicate is verified.
+
+     If `pred` verifies the given `attrs` set, the result is a list with a
+     single element `f [ ] attrs`.
```

Comment:
Order of recursion? Looks like it traverse the leafs first then going up? Sometimes this can be important to know.